astroid/decorators.py

```diff
@@ -152,58 +152,79 @@ def raise_if_nothing_inferred(func, instance, args, kwargs):
     yield from generator
 
 
-def deprecate_default_argument_values(
-    astroid_version: str = "3.0", **arguments: str
-) -> Callable[[Callable[P, R]], Callable[P, R]]:
-    """Decorator which emitts a DeprecationWarning if any arguments specified
-    are None or not passed at all.
-
-    Arguments should be a key-value mapping, with the key being the argument to check
-    and the value being a type annotation as string for the value of the argument.
-    """
-    # Helpful links
-    # Decorator for DeprecationWarning: https://stackoverflow.com/a/49802489
-    # Typing of stacked decorators: https://stackoverflow.com/a/68290080
-
-    def deco(func: Callable[P, R]) -> Callable[P, R]:
-        """Decorator function."""
-
-        @functools.wraps(func)
-        def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
-            """Emit DeprecationWarnings if conditions are met."""
-
-            keys = list(inspect.signature(func).parameters.keys())
-            for arg, type_annotation in arguments.items():
-                try:
-                    index = keys.index(arg)
-                except ValueError:
-                    raise Exception(
-                        f"Can't find argument '{arg}' for '{args[0].__class__.__qualname__}'"
-                    ) from None
-                if (
-                    # Check kwargs
-                    # - if found, check it's not None
-                    (arg in kwargs and kwargs[arg] is None)
-                    # Check args
-                    # - make sure not in kwargs
-                    # - len(args) needs to be long enough, if too short
-                    #   arg can't be in args either
-                    # - args[index] should not be None
-                    or arg not in kwargs
-                    and (
-                        index == -1
-                        or len(args) <= index
-                        or (len(args) > index and args[index] is None)
-                    )
-                ):
-                    warnings.warn(
-                        f"'{arg}' will be a required argument for "
-                        f"'{args[0].__class__.__qualname__}.{func.__name__}' in astroid {astroid_version} "
-                        f"('{arg}' should be of type: '{type_annotation}')",
-                        DeprecationWarning,
-                    )
-            return func(*args, **kwargs)
-
-        return wrapper
-
-    return deco
+# Expensive decorators only used to emit Deprecation warnings.
+# If no other than the default DeprecationWarning are enabled,
+# fall back to passthrough implementations.
+if util.check_warnings_filter():
```

Comment:
Nitpick: I think we could remove some duplication by putting this if inside ``deprecate_default_argument_values``

Reply:
I would suggest not to do that. The way it's written currently, the check is only done once when the module is loaded initially. If we move it inside the function, it's executed once per call.

Reply:
Right, I did not realize that ! Resolving.